src/core/webidl.js

```diff
@@ -358,6 +359,38 @@ function renderWebIDL(idlElement, index) {
   addIDLHeader(idlElement);
   return parse;
 }
+
+/**
+ * Add data-dfn-for to the closest section if not present already.
+ * @param {HTMLPreElement} idlElement
+ * @param {ReturnType<typeof webidl2.parse>} parse
+ */
+function addDataDfnFor(idlElement, parse) {
+  const closestSection = idlElement.closest("section");
+  if (closestSection.hasAttribute("data-dfn-for")) return;
+
+  const topLevelEntities = ["dictionary", "interface", "callback interface"];
```

Comment:
Need "interface mixin" also, right? Or do we want to support only "dictionary" and "interface"?
https://github.com/w3c/respec/blob/f62ee6b95d79c6008d01da35a95464b5854b870b/src/core/webidl.js#L240-L244

Reply:
yeah, it would be all the top level ones...